### ISSUE TITLE
Support WinRAR 5.x SFX files

### ISSUE THREAD

**Issue by kholia** [magnumripper/JohnTheRipper#2748]
http://www.openwall.com/lists/john-users/2017/09/07/1

**Comment by kholia** [comment]
Here is patch to fix this problem,

``` diff
diff --git a/src/rar2john.c b/src/rar2john.c
index a114e0d..1f3a271 100644
--- a/src/rar2john.c
+++ b/src/rar2john.c
@@ -214,6 +214,7 @@ static void process_file(const char *archive_name)
                if (memcmp(marker_block, "MZ", 2) == 0) {
                        /* jump to "Rar!" signature */
                        while (!feof(fp)) {
                                count = fread(buf, 1, CHUNK_SIZE, fp);
                                if ( (pos = memmem(buf, count, "\x52\x61\x72\x21\x1a\x07\x00", 7))) {
                                        diff = count - (pos - buf);
                                        jtr_fseek64(fp, - diff, SEEK_CUR);
                                        jtr_fseek64(fp, 7, SEEK_CUR);
                                        found = 1;
                                        break;
                                }
                                if (feof(fp)) //We shold examine the EOF before seek back
                                        break;
                                jtr_fseek64(fp, -6, SEEK_CUR);
                        }
                        if (!found) {
+                               if (process_file5(archive_name))
+                                       return;
                                fprintf(stderr, "! %s: Not a RAR file\n", archive_name);
                                goto err;
                        }
                }
                else {
                        /* try to detect RAR 5 files */
                        fclose(fp);
                        fp = NULL;
                        MEM_FREE(best);
                        MEM_FREE(gecos);
                        if (process_file5(archive_name))
                                return;
                        fprintf(stderr, "! %s: Not a RAR file\n", archive_name);
                        goto err;
                }
        }

```

Currently, the WinRAR 3.x SFX handling code never calls `process_files5` function. This behavior is not correct.

**Comment by kholia** [comment]
Fixed in PR #2749.
